cocotb/handle.py

```diff
@@ -625,7 +655,12 @@ def setimmediatevalue(self, value):
             self._log.critical("Unsupported type for value assignment: %s (%s)", type(value), repr(value))
             raise TypeError("Unable to set simulator value with type %s" % (type(value)))
 
-        simulator.set_signal_val_binstr(self._handle, value.binstr)
+        simulator.set_signal_val_binstr(self._handle, set_action, value.binstr)
+
+    def check_for_set_action(self, value):
```

Comment:
```suggestion
    def _check_for_set_action(self, value):
```
so as not to be public API. Not a huge fan of this name, but I don't have a better one to hand.

Reply:
I feel the same, but don't have a better name either. Sorry!